src/Umbraco.Core/Manifest/ManifestContentAppDefinition.cs

```diff
@@ -109,23 +112,49 @@ public ContentApp GetContentAppFor(object o)
             {
                 var ok = false;
 
-                // else iterate over each entry
-                foreach (var rule in rules)
+                //if any role specific rules, deal with them first.
+                if (rules.Where(x => x.PartA.InvariantEquals("role")).Any())
```

Comment:
Not annoying at all @zpqrtbnk, it's always good to see how it can be improved :) I am happy to run through all the test scenarios again once done.

The PR says "open", is that what you mean?

Reply:
no idea ;-) what I mean is whatever is needed to let me push to the branch - I think Sebastiaan knows what I am talking about but I cannot remember what it is exactly!

Reply:
@crgrieve When viewing the *Conversation* tab of your PR, there is a sidebar to the right with a checkbox for he sidebar to the right for allowing edits from maintainers.

![image](https://user-images.githubusercontent.com/3634580/46951615-cb50cd00-d088-11e8-964f-72de581d0a69.png)

Reply:
Thanks @abjerner! Yeah, that's selected. Hopefully you can edit @zpqrtbnk?